src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/GetDownloadResultUtility.cs

```diff
@@ -51,12 +57,25 @@ public static class GetDownloadResultUtility
                                 return new DownloadResourceResult(DownloadResourceResultStatus.NotFound);
                             }
 
-                            return await GlobalPackagesFolderUtility.AddPackageAsync(
-                                identity,
-                                packageStream,
-                                settings,
-                                logger,
-                                token);
+                            if (cacheContext.NoCache)
+                            {
+                                return await AddPackageDirectAsync(
+                                    identity,
+                                    packageStream,
+                                    settings,
+                                    cacheContext.GeneratedTempFolder,
+                                    logger,
+                                    token);
+                            }
+                            else
+                            {
+                                return await GlobalPackagesFolderUtility.AddPackageAsync(
```

Comment:
We have two blocks of code each of which do something slightly different:
- InstallFromSourceAsync is invoked with a VersionFolderPathContext initialized with either the PackagesDirectory or globalPackagesFolder
- And then the DownloadResourceResult is either constructed from the globalPackagesFolder or the PackagesDirectory which requires a slightly different approach.
We can either keep the different approaches in separate methods as I chose to do or create one method with conditional logic for each branch.